packages/alfa-json/src/serializable.ts

```diff
@@ -20,6 +20,12 @@ export interface Serializable<T extends JSON = JSON> {
 }
 
 export namespace Serializable {
+  export type ToJSON<T> = T extends Serializable<infer T>
+    ? T
+    : T extends JSON
+    ? T
+    : JSON;
+
```

Comment:
Wait, are there two `T` here? One introduced by `ToJSON<T>` and one by `infer T`.
We should name them differently, especially since both `? T` then correspond to different `T` (if I understand correctly), which is definitely confusing…